lir/visualization.py

```diff
@@ -1,38 +1,39 @@
+from collections.abc import Callable
 from pathlib import Path
-
-import numpy as np
+from typing import Any
 
 from lir.data.models import LLRData
 from lir.plotting import savefig
 
 
-def pav(base_path: Path, llrdata: LLRData, labels: np.ndarray) -> None:
-    """Helper function to generate and save a PAV-plot to the output directory."""
-    base_path.mkdir(exist_ok=True, parents=True)
-    path = base_path / 'pav.png'
-    with savefig(str(path)) as fig:
-        fig.pav(llrdata, labels)
+def _save_or_plot(ax: Any | None, base_path: Path | None, filename: str, plot_func: Callable, *args: Any) -> None:
+    """Internal helper to either plot to an axis or save a plot to file."""
+    if ax is not None:
+        plot_func(ax, *args)
+    elif base_path is not None:
+        base_path.mkdir(exist_ok=True, parents=True)
+        path = base_path / filename
+        with savefig(str(path)) as fig:
+            plot_func(fig, *args)
+    else:
+        raise ValueError('Either base_path or ax must be provided.')
+
+
+def pav(base_path: Path | None, llrdata: LLRData, ax: Any | None = None) -> None:
```

Comment:
Deze plottingfunctie (en de rest van deze file) was al een adapter voor `lir.plotting`. Nu we ook nog een aggregation-laag ertussen hebben voegt deze laag vooral complexiteit toe. (Kan ook in de volgende PR, maar maak dan een issue aan want het moet eigenlijk wel gebeuren.)